modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/core/MultiListenerServerIODispatch.java

```diff
@@ -46,32 +49,59 @@ public MultiListenerServerIODispatch(final Map<Integer, NHttpServerEventHandler>
     protected void onConnected(final DefaultNHttpServerConnection defaultNHttpServerConnection) {
         int localPort = defaultNHttpServerConnection.getLocalPort();
         NHttpServerEventHandler handler = handlers.get(localPort);
-        try {
-            handler.connected(defaultNHttpServerConnection);
-        } catch (final Exception ex) {
-            handler.exception(defaultNHttpServerConnection, ex);
+        if (handler == null) {
+            log.error("Listener has not started yet on " + localPort + ". Hence shutting down the connection");
+            try {
+                defaultNHttpServerConnection.shutdown();
+            } catch (IOException ex) {
+                log.error("Error shutting down the connection", ex);
```

Comment:
without logging an error message, shall we simply ignore the exception as in https://github.com/wso2/wso2-synapse/blob/d66dd9a13362d4b998378779391d22f6223b9088/modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/connections/SourceConnections.java#L157

Reply:
Done